src/jsonrpc/api.rs

```diff
@@ -0,0 +1,87 @@
+use crate::jsonrpc::types::{MpoolPushMessage, MpoolPushMessageResponse};
+use crate::jsonrpc::JsonRpcClient;
+use anyhow::{anyhow, Result};
+use cid::Cid;
+use fvm_shared::econ::TokenAmount;
+use num_traits::cast::ToPrimitive;
+use serde::de::DeserializeOwned;
+use serde_json::json;
+use std::collections::HashMap;
+
+const DEFAULT_VERSION: u16 = 42;
+const MESSAGE_KEY: &str = "Message";
+
+// RPC endpoints
+const MEM_PUSH_MESSAGE_ENDPOINT: &str = "Filecoin.MpoolPushMessage";
+
+pub struct NodeApi<Inner: JsonRpcClient> {
+    inner: Inner,
+}
+
+impl<Inner: JsonRpcClient> NodeApi<Inner> {
+    pub fn new(inner: Inner) -> Self {
+        Self { inner }
+    }
+
+    pub async fn mpool_push_message(&self, msg: MpoolPushMessage) -> Result<Cid> {
```

Comment:
I think `mpool_push_message` is better, since we are changing the struct to `LotusAPI`, then this is lotus specific, should be aligned with lotus api endpoint is easier to read. Also the function is declared as `async`, so I guess should be no need to specify `async`